test/e2e/node/mount_propagation.go

```diff
@@ -0,0 +1,178 @@
+	. "github.com/onsi/gomega"
+)
+
+func preparePod(name string, node *v1.Node, propagation v1.MountPropagationMode, hostDir string) *v1.Pod {
+	const containerName = "cntr"
+	bTrue := true
+	var oneSecond int64 = 1
+	// The pod prepares /mnt/test/<podname> and sleeps.
+	cmd := fmt.Sprintf("mkdir /mnt/test/%[1]s; sleep 3600", name)
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1.PodSpec{
+			NodeName: node.Name,
+			Containers: []v1.Container{
+				{
+					Name:    containerName,
+					Image:   "busybox",
+					Command: []string{"sh", "-c", cmd},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:             "host",
+							MountPath:        "/mnt/test",
+							MountPropagation: &propagation,
+						},
+					},
+					SecurityContext: &v1.SecurityContext{
+						Privileged: &bTrue,
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: "host",
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{
+							Path: hostDir,
+						},
+					},
+				},
+			},
+			// speed up termination of the pod
+			TerminationGracePeriodSeconds: &oneSecond,
+		},
+	}
+	return pod
+}
+
+var _ = SIGDescribe("Mount propagation [Feature:MountPropagation]", func() {
+	f := framework.NewDefaultFramework("mount-propagation")
+
+	It("should propagate mounts to the host", func() {
+		// This test runs two pods: master and slave with respective mount
+		// propagation on common /var/lib/kubelet/XXXX directory. Both mount a
+		// tmpfs to a subdirectory there. We check that these mounts are
+		// propagated to the right places.
+
+		// Pick a node where all pods will run.
+		nodes := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		Expect(len(nodes.Items)).NotTo(BeZero(), "No available nodes for scheduling")
+		node := &nodes.Items[0]
+
+		// hostDir is the directory that's shared via HostPath among all pods.
+		// Make sure it's random enough so we don't clash with another test
+		// running in parallel.
+		hostDir := "/var/lib/kubelet/" + f.Namespace.Name
+		defer func() {
+			cleanCmd := fmt.Sprintf("sudo rm -rf %q", hostDir)
+			framework.IssueSSHCommand(cleanCmd, framework.TestContext.Provider, node)
+		}()
+
+		podClient := f.PodClient()
+		master := podClient.CreateSync(preparePod("master", node, v1.MountPropagationBidirectional, hostDir))
+		defer podClient.Delete(master.Name, nil)
+
+		slave := podClient.CreateSync(preparePod("slave", node, v1.MountPropagationHostToContainer, hostDir))
+		defer podClient.Delete(slave.Name, nil)
+
+		// Check that the pods sees directories of each other. This just checks
+		// that they have the same HostPath, not the mount propagation.
+		podNames := []string{master.Name, slave.Name}
+		for _, podName := range podNames {
+			for _, dirName := range podNames {
+				cmd := fmt.Sprintf("test -d /mnt/test/%s", dirName)
+				_ = f.ExecShellInPod(podName, cmd)
```

Comment:
Are you sure this actually captures a non-zero exit code? (try flipping the test and verify that it fails)

Reply:
Yes, it leads to [ExecCommandInContainer](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/exec_util.go#L104) that prints error:

```
• Failure [36.984 seconds]
[k8s.io] [sig-node] Mount propagation [Feature:MountPropagation]
/home/jsafrane/project/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:642
  should propagate mounts to the host [It]
  /home/jsafrane/project/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/node/mount_propagation.go:80

  failed to execute command in pod master, container cntr: command terminated with exit code 1
  Expected error:
      <exec.CodeExitError>: {
          Err: {
              s: "command terminated with exit code 1",
          },
          Code: 1,
      }
      command terminated with exit code 1
  not to have occurred
```